fvm/evm/types/call.go

```diff
@@ -0,0 +1,131 @@
+package types
+
+import (
+	"math/big"
+
+	gethCommon "github.com/ethereum/go-ethereum/common"
+	gethCore "github.com/ethereum/go-ethereum/core"
+	gethCrypto "github.com/ethereum/go-ethereum/crypto"
+	"github.com/ethereum/go-ethereum/rlp"
+)
+
+const (
+	// tx type 255 is used for direct calls from bridged accounts
+	DirectCallTxType = byte(255)
+
+	UnknownCallSubType  = byte(0)
+	DepositCallSubType  = byte(1)
+	WithdrawCallSubType = byte(2)
+	TransferCallSubType = byte(3)
+	DeployCallSubType   = byte(4)
+	ContractCallSubType = byte(5)
+
+	TransferGasUsage = 21_000
+)
+
+// DirectCall captures all the data related to a direct call to evm
+// direct calls are similar to transactions but they don't have
+// signatures and don't need sequence number checks
+type DirectCall struct {
+	Type     byte
+	SubType  byte
+	From     Address
+	To       Address
+	Data     []byte
+	Value    *big.Int
+	GasLimit uint64
+}
+
+// Encode encodes the direct call it also adds the type
+// as the very first byte, similar to how evm encodes types.
+func (dc *DirectCall) Encode() ([]byte, error) {
+	encoded, err := rlp.EncodeToBytes(dc)
+	return append([]byte{dc.Type}, encoded...), err
+}
+
+// Hash computes the hash of a direct call
+func (dc *DirectCall) Hash() (gethCommon.Hash, error) {
+	encoded, err := dc.Encode()
+	return gethCrypto.Keccak256Hash(encoded), err
+}
+
+// Message constructs a core.Message from the direct call
+func (dc *DirectCall) Message() *gethCore.Message {
+	var to *gethCommon.Address
+	if dc.To != EmptyAddress {
+		ct := dc.To.ToCommon()
+		to = &ct
+	}
+	return &gethCore.Message{
+		From:      dc.From.ToCommon(),
+		To:        to,
+		Value:     dc.Value,
+		Data:      dc.Data,
+		GasLimit:  dc.GasLimit,
+		GasPrice:  big.NewInt(0), // price is set to zero fo direct calls
+		GasTipCap: big.NewInt(1), // also known as maxPriorityFeePerGas
+		GasFeeCap: big.NewInt(2), // also known as maxFeePerGas
```

Comment:
Hardcoding this to 2 will fail if the base fee gets changed, which is possible since it's configurable. Should we just let it auto calculate it?

Reply:
this requires more work on details, for now I removed the option to configure base fee but we can revisit later 
[48be317](https://github.com/onflow/flow-go/pull/4927/commits/48be31750a6b4634d4e39f186cf60204d0032ce1)